data/libraries.js

```diff
@@ -1,127 +1,160 @@
-        // ejinks - reconfiged to export all
-        id = validation.checkValidId(id, "Library ID");
-        const libraryCollection = await libraries();
-        let library = await libraryCollection.findOne({_id: new ObjectId(id)});
-        if (library === null) throw "Error: No library found with given ID.";
-        library._id = library._id.toString();
-        return library;
-    },
-    async getAllComments(id) {
-        // ejinks
-        id = validation.checkValidId(id, "Library ID");
-        const libraryCollection = await libraries();
-        let commentsList = await libraryCollection.findOne(
-            {_id: new ObjectId(id)},
-            {_id: 0, comments: 1}
-        );
-        if (commentsList == null) throw "Error: No library found with given ID.";
-        return commentsList;
-    },
-    async createComment(libraryId, userId, text) {
-        // ejinks
-        libraryId = validation.checkValidId(libraryId, "Library ID");
-        userId = validation.checkValidId(userId, "User ID");
+  async create(
+    name,
+    location, // Vish will help!! possibly using the API?
+    image, // Vish will help!!
+    ownerID,
+    fullnessRating,
+    genres
+  ) {
+    name = validation.checkString(name, "Library Name");
+    ownerID = validation.checkId(ownerID, "Library Owner ID");
+    fullnessRating = validation.isValidNumber(
+      fullnessRating,
+      "Fullness Rating"
+    );
+    genres = validation.checkStringArray(genres, "Genres Available");
+    const currentDate = new Date();
+    const lastServayed = currentDate.toLocaleString(undefined, {
+      // should be in form "7/22/2016, 04:21 AM"
+      day: "numeric",
+      month: "numeric",
+      year: "numeric",
+      hour: "2-digit",
+      minute: "2-digit",
+    });
+    let newLibrary = {
+      name: name,
+      location: location,
+      image: image,
+      ownerID: ownerID,
+      fullnessRating: fullnessRating,
+      lastServayed: lastServayed,
+      genres: genres,
+      favorites: [],
+      comments: [],
+    };
+    const librariesCollection = await libraries();
+    const insertInfo = await librariesCollection.insertOne(newLibrary);
+    if (!insertInfo.acknowledged || !insertInfo.insertedId) {
+      throw "Error: Could not add Library";
+    }
+    insertInfo.insertedId = insertInfo.insertedId.toString();
+    let res = await this.getLibraryById(insertInfo["insertedId"].toString());
+    return res;
+  },
+  /** This function gets all the libraries and returns the json data in an array*/
+  async getAllLibraries() {
+    const librariesCollection = await libraries();
+    return await librariesCollection.find({}).toArray();
+  },
+  async get(id) {
+    // ejinks - reconfiged to export all
+    id = validation.checkValidId(id, "Library ID");
+    const libraryCollection = await libraries();
+    let library = await libraryCollection.findOne({ _id: new ObjectId(id) });
+    if (library === null) throw "Error: No library found with given ID.";
```

Comment:
@JulienCarr2  @DanielleDauphinais  @galapatt  @ejinks 
Just a small coding practice to make code more readable by adding a new line before any control statements like 
if or for or an await

Reply:
if you don't like this coding style, you can ignore it